### PR TITLE
clist: add a few basic clist tests

### PR DESCRIPTION
**Pull request by williambanfield**

### CODE REVIEW

File: internal/libs/clist/clist_test.go [hunk 1]
```diff
@@ -341,3 +342,63 @@ FOR_LOOP2:
...
+	l.Remove(el1)
+	require.True(t, el1.Removed())
+	require.False(t, el2.Removed())
+
+	l.Clear()
+	require.True(t, el2.Removed())
+	require.True(t, el3.Removed())
+}
+
+func TestNextWaitChan(t *testing.T) {
```
Comment (tychoish) [inline internal/libs/clist/clist_test.go:360]: can we do this with `require.Eventually` and also subtests?

**Review by tychoish — CHANGES_REQUESTED** [review]

File: internal/libs/clist/clist_test.go (hunk 1 above)
Comment (williambanfield) [reply internal/libs/clist/clist_test.go:360]: Changed to subtests. Realized test was not async, so switched to `default` from `time.After`.

**Review by tychoish — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]
Nice! 

I'm surprised that there's not better testing support for using channels

**Comment by williambanfield** [comment]
> Nice!
> 
> I'm surprised that there's not better testing support for using channels

I generally try to avoid exposing channels in APIs tbh. I find them a little difficult to reason about and test.

**Comment by tychoish** [comment]
> I generally try to avoid exposing channels in APIs tbh. I find them a little difficult to reason about and test.

Hard concur.

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**